cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb

```diff
@@ -0,0 +1,40 @@
+resource_name :munge_key_manager
+provides :munge_key_manager
+unified_mode true
+
+property :munge_key_secret_arn, String
+
+default_action :manage
+
+action :manage do
+  if new_resource.munge_key_secret_arn
+    # This block will fetch the munge key from Secrets Manager
+    bash 'fetch_and_decode_munge_key' do
+      user 'root'
+      group 'root'
+      cwd '/tmp'
+      code <<-FETCH_AND_DECODE
+        # Get encoded munge key from secrets manager and decode it
+        encoded_key=$(aws secretsmanager get-secret-value --secret-id #{new_resource.munge_key_secret_arn} --query 'SecretString' --output text)
```

Comment:
What if there's an error fetching the secret? This will not return the value.
Can you check if `encoded_key` has a value and possibly echo and error & exit early before base64 decoding it?